File: FRONTEND/src/services/taskService.js
import api from './api';

export const getTasksByUser = async (userId) => {
  const response = await api.get(`/api/tasks/user/${userId}`);
  return response.data;
};

export const createTask = async (taskData) => {
  const response = await api.post('/api/tasks/create', taskData);
  return response.data;
};

export const updateTaskStatus = async (id, status) => {
  const response = await api.patch(`/api/tasks/${id}/status`, null, {
    params: { status }
  });
  return response.data;
};

// Putaway operations
export const getPutawayItems = async (userId) => {
  const response = await api.get(`/api/tasks/putaway/user/${userId}`);
  return response.data;
};

export const getPutawayStatistics = async (userId) => {
  const response = await api.get(`/api/tasks/putaway/statistics/${userId}`);
  return response.data;
};

export const getRecentCompletions = async (userId) => {
  const response = await api.get(`/api/tasks/putaway/recent-completions/${userId}`);
  return response.data;
};

export const startPutaway = async (taskId) => {
  const response = await api.post(`/api/tasks/${taskId}/start-putaway`);
  return response.data;
};

export const completePutaway = async (taskId, binId, quantity) => {
  const response = await api.post(`/api/tasks/${taskId}/complete-putaway`, {
    binId,
    quantity
  });
  return response.data;
};

export const completePutawayWithAllocation = async (taskId, allocations) => {
  const response = await api.post(`/api/tasks/${taskId}/complete-putaway`, {
    allocations
  });
  return response.data;
};

// Picking operations
export const getPickingItems = async (userId) => {
  const response = await api.get(`/api/tasks/picking/user/${userId}`);
  return response.data;
};

export const getAssignedPickingItems = async (userId) => {
  const response = await api.get(`/api/tasks/picking/assigned/${userId}`);
  return response.data;
};

export const getPickingStatistics = async (userId) => {
  const response = await api.get(`/api/tasks/picking/statistics/${userId}`);
  return response.data;
};

export const getDispatchedPickingItems = async (userId) => {
  const response = await api.get(`/api/tasks/picking/dispatched/${userId}`);
  return response.data;
};

export const completePicking = async (taskId, userId) => {
  const response = await api.post(`/api/tasks/${taskId}/complete-picking`, null, {
    params: { userId }
  });
  return response.data;
};

File: BACKEND/backend/src/main/java/com/visera/backend/Service/TaskService.java
package com.visera.backend.Service;
import com.visera.backend.DTOs.BinAllocation;
import com.visera.backend.DTOs.PickingStatisticsDTO;
import com.visera.backend.DTOs.PutawayStatisticsDTO;
import com.visera.backend.DTOs.RecentCompletionDTO;
import com.visera.backend.Entity.Task;
import java.util.List;

public interface TaskService {
    Task createTask(Task task);
    List<Task> getTasksByUser(int userId);
    Task updateTaskStatus(int id, String status);
    List<Task> getPutawayTasksByUser(int userId);
    Task startPutaway(Long taskId);
    List<Task> getCompletedPutawayTasksToday(int userId);
    PutawayStatisticsDTO getPutawayStatistics(int userId);
    Task completePutaway(Long taskId, Long binId, Integer quantity);
    Task completePutawayWithAllocation(Long taskId, List<BinAllocation> allocations);
    
    // Picking operations
    List<Task> getPickingTasksByUser(int userId);
    List<Task> getAllPickingTasksForViewing(int userId);
    List<Task> getDispatchedPickingTasks(int userId);
    PickingStatisticsDTO getPickingStatistics(int userId);
    Task completePicking(Long taskId, int userId);
}
